html/js/graph.js

```diff
@@ -0,0 +1,1044 @@
+// Declaration of d3 variable (cf. d3js)
+var d3;
+
+// **************
+// constants_endpoints
+// **************
+// Endpoints (cf. REST endpoints in App.py)
+var URL_ROOT_API = "https://tuttifrutti.alwaysdata.net/";
+var ENDPOINT_STORES = "/fetchStores/";
+var ENDPOINT_SCORE_DBS = "/fetchScoreDbs";
+
+/*
+ * purpose: Constants related to template page stats.html for displaying the statistics of database_aggregation into a table
+ */
+
+// json properties for each database
+var FLD_DB_DISPLAY_NAME = "dbDisplayName";
+var FLD_DB_NICK_NAME = "dbNickname";
+
+// **************
+// constants
+// **************
+
+// Minimum proximity of matching products with reference-product for being part of suggestions
+var MAX_STORES_TO_SHOW_PER_COUNTRY = 100;
+var MIN_SCORE_FOR_SUGGESTIONS = 70;
+var MAX_SUGGESTIONS = 50;
+// popup messages
+var MSG_WAITING_SCR_FETCH_STORES = ".. fetching stores ..";
+var MSG_WAITING_SCR_MATCH_REQUEST = ".. please wait ..";
+
+// url parameters (optional)
+var URL_PARAM_BARCODE="barcode";
+var URL_PARAM_COUNTRY="country";
+var URL_PARAM_SCORE="score";
+
+var GRAPH_WIDTH = $(window).innerWidth() * 75 / 100;
```

Comment:
Fixed sizes for the graph and similar visual elements don't really work if you resize a window (ie. full screen to half screen width) or rotate a phone. It would be better if this was responsive to window size changes etc.

Reply:
ouhla ... that reaches my knowledge-limits I am afraid! Can someone help?